tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -68,6 +69,32 @@ async def discover_module_values(dut):
         raise TestFailure("Expected to discover things in the DUT")
 
 
+@cocotb.test(skip=True)
+def ipython_embed(dut):
+    yield Timer(0)
+    import IPython
+    IPython.embed()
+
+
+@cocotb.test(skip=True)
+def ipython_embed_kernel(dut):
+    """Start an interactive Python shell."""
+    yield Timer(0)
+    import IPython
+    print(textwrap.dedent("""
+    ###############################################################################
+    Running IPython embed_kernel()
+
+    You can now send this process into the background with "Ctrl-Z bg" and run
+        jupyter console --existing
+    or
+        jupyter qtconsole --existing
+    or
+        jupyter console --existing kernel-{}.json
+    ###############################################################################""".format(os.getpid())))
+    IPython.embed_kernel()
+
+
```

Comment:
These were removed in another PR, we don't want them back.
```suggestion
```

Reply:
wasn't sure. but we also need to clean up imports.